legend-engine-shared-core/src/main/java/org/finos/legend/engine/shared/core/api/grammar/GrammarAPI.java

```diff
@@ -57,6 +58,23 @@ protected <T> Response grammarToJson(String text, Function<String, T> func, Prof
         }
     }
 
+    protected <T> Response grammarToJson(Map<String, String> text, Function<Map<String, String>, T> func, ProfileManager<CommonProfile> pm, String spanText)
+    {
+        MutableList<CommonProfile> profiles = ProfileManagerHelper.extractProfiles(pm);
+        try (Scope scope = GlobalTracer.get().buildSpan(spanText).startActive(true))
+        {
+            try
+            {
+                T data = func.apply(text);
+                return ManageConstantResult.manageResult(profiles, data, objectMapper);
+            }
+            catch (Exception e)
+            {
+                return ExceptionTool.exceptionManager(e, LoggingEventType.TRANSFORM_GRAMMAR_TO_JSON_ERROR, Response.Status.BAD_REQUEST, profiles);
+            }
+        }
+    }
+
     protected <T> Response grammarToJsonBatch(Map<String, ParserInput> input, Function5<String, String, Integer, Integer, Boolean, T> func, Map<String, T> result, ProfileManager<CommonProfile> pm, String spanText)
```

Comment:
why can't we reuse this method for our purpose? after all the input and output object shapes look right to me

Reply:
We can't really use this function. As the output of function would return json for each grammar element. It would be of format Map<key, PMCD>. But that's not something we would want for our use case. If we receive result of this format we won't be able to send this as an input to compilation in studio. We need to join everthing before sending it as an input.
Another disadvantage is we get sectionIndex for every grammar element which is something we won't be able to compile.